contracts/PositionManager.sol

```diff
@@ -0,0 +1,163 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.20;
+
+import { IPositionManager } from "./interfaces/IPositionManager.sol";
+import { DoubleEndedQueue } from "openzeppelin/utils/structs/DoubleEndedQueue.sol";
+
+contract PositionManager is IPositionManager {
+    using DoubleEndedQueue for DoubleEndedQueue.Bytes32Deque;
+
+    // TODO: Reassess using a more tailored data structure.
+    /// @dev Utility data structure to manage the position queue.
+    ///      Supports pushing and popping from both the front and back.
+    DoubleEndedQueue.Bytes32Deque private _positions;
+
+    /// @dev Last checkpoint time the portfolio was rebalanced.
+    uint256 private _lastRebalancedTimestamp;
+
+    /// @inheritdoc IPositionManager
+    function getPositionCount() public view returns (uint256) {
+        return _positions.length();
+    }
+
+    /// @inheritdoc IPositionManager
+    function getPosition(
+        uint256 _index
+    ) public view returns (Position memory position) {
+        position = _decodePosition(_positions.at(_index));
+    }
+
+    /// @inheritdoc IPositionManager
+    function hasMaturedPositions() public view returns (bool) {
+        // Return false if there are no positions.
+        if (_positions.length() == 0) return false;
+        Position memory _position = _decodePosition(_positions.at(0));
+
+        // Return true if the current block timestamp is after
+        // the oldest position's `maturityTime`.
+        if (_position.maturityTime <= block.timestamp) {
+            return true;
+        }
+
+        // Return false since the oldest position has not matured.
+        return false;
```

Comment:
we could just return _position.maturityTime <= block.timestamp, but I would understand for extended readability

Reply:
derp you so right ty